src/ServiceManagement/Automation/Commands.Automation/Cmdlet/SetAzureAutomationVariable.cs

```diff
@@ -31,21 +32,29 @@ public class SetAzureAutomationVariable : AzureAutomationBaseCmdlet
         /// <summary>
         /// Gets or sets the variable name.
         /// </summary>
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The variable name.")]
+        [Parameter(ParameterSetName = AutomationCmdletParameterSets.UpdateVariableValue, Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The variable name.")]
+        [Parameter(ParameterSetName = AutomationCmdletParameterSets.UpdateVariableDescription, Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The variable name.")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         /// <summary>
+        /// Gets or sets the variable encrypted Property.
+        /// </summary>
+        [Parameter(ParameterSetName = AutomationCmdletParameterSets.UpdateVariableValue, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The encrypted property of the variable.")]
+        [ValidateNotNull]
+        public bool Encrypted { get; set; }
+        
+        /// <summary>
         /// Gets or sets the variable description.
         /// </summary>
-        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The description of the variable.")]
+        [Parameter(ParameterSetName = AutomationCmdletParameterSets.UpdateVariableDescription, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The description of the variable.")]
```

Comment:
does this look right? why value is mandatory for updatevariablevalue parameterset but description not mandatory for updatevariabledescription parameterset?